specs/trust-store-trust-policy.md

```diff
@@ -81,9 +81,31 @@ Users who consume signed artifacts from OCI registries, or signed arbitrary blob
+  - **`trustStores`**(*array of string*): This REQUIRED property specifies a set of one or more named trust stores, each of which contain the trusted roots against which signatures are verified. Each named trust store uses the format `{trust-store-type}:{named-store}`. Currently supported values for `trust-store-type` are `ca`, `signingAuthority` and `tsa`. When a TSA is used, the format `ca:acme-rockets,tsa:acme-tsa` is supported to specify it.
   - **`trustedIdentities`**(*array of strings*): This REQUIRED property specifies a set of identities that the user trusts. For X.509 PKI, it supports list of elements/attributes of the signing certificate's subject. For more information, see [identities constraints](#trusted-identities-constraints) section. A value `*` is supported if user trusts any identity (signing certificate) issued by the CA(s) in `trustStore`.
 
+##### Version 2.0
+Version 2.0 adds support of timestamp countersignature verification in addition to version 1.0. Version 2.0 also provides trust source support for `trustedIdentities` and optimizes the format of `trustStores` to match the upgraded `trustedIdentities` field.
+
+- **`version`**(*string*): This REQUIRED property is the version of the trust policy. This value MUST be `2.0` for version 2.0 trust policy.
+- **`trustPolicies`**(*string-array of objects map*): This REQUIRED property represents a collection of trust policies.
+  - **`name`**(*string*): This REQUIRED property represents the name of the trust policy.
+  - **`registryScopes`**(*array of strings*): This REQUIRED property determines which trust policy is applicable for the given artifact.
+    The scope field supports filtering based on fully qualified repository URI `${registry-name}/${namespace}/${repository-name}`.
+    For more information, see [trust policy constraints](#oci-trust-policy-constraints) section.
+  - **`signatureVerification`**(*object*): This REQUIRED property dictates how signature verification is performed.
+  An *object* that specifies a predefined verification level, with an option to override the Notary Project trust policy defined verification level if user wants to specify a [custom verification level](#custom-verification-level).
+    - **`level`**(*string*): A REQUIRED property that specifies the verification level, supported values are `strict`, `permissive`, `audit` and `skip`. Detailed explanation of each level is present [here](#signatureverification-details).
+    - **`override`**(*map of string-string*): This OPTIONAL map is used to specify a [custom verification level](#custom-verification-level).
+  - **`timestampVerification`**(*object*): This OPTIONAL property dictates how timestamp countersignature verification is performed. If present, it is used solely under the [`notary.x509`](./signing-scheme.md/#notaryx509) signing scheme. If not present, default timestamp countersignature verification will be performed. Current supported fields are:
```

Comment:
Why do we need `timestampVerification` ?
* `level`: Can this be governed by `signatureVerification#level` ?
* `expiryRelaxed`: As per spec, we dont care about expiry of timestamping leaf cert, then why do we need this field?

Reply:
@priteshbandi suggested in the community meeting that not configuring any tsa trust store iff `level` to be `skip` so that we don’t need to introduce the field `level`.

Reply:
@shizhMSFT  @priteshbandi @yizha1 I just watched the Monday community meeting's recording, looks like I missed a really good discussion :) 
I notice that for some of the topics discussed in the meeting, they are actually already covered in this PR, for example, [Timestamp Countersignature Verification details](https://github.com/notaryproject/specifications/blob/2a6833fb66af44c3f974b2abaceedd6b543bcc49/specs/trust-store-trust-policy.md#timestamp-countersignature-verification-details).
So, my following explanation will be based on you have already read through this PR or at least the `trust-store-trust-policy.md` :) And I really appreciate your time as this is indeed a major change.
1.  The reason to have a separate `timestampVerification` config rather than mixing it within `Authentic timestamp` is that timestamp is just one type of counter signature. We could have other types of counter signatures in the future, mixing all of them into `signature verification` makes the trust policy really hard to understand. In our case, the `Authentic timestamp` field, as an overall switch, controls the signing cert validity check for both of our signing schemes, and we only support timestamp verification for one of them (i.e., `notary.x509`). Mixing them makes the trust policy hard to understand.
2. All roads lead to Rome. IMHO, explicitly showing the configuration of timestamp verification makes the trust policy much more straightforward. Checking existence of `tsa` trust store can also do the job, but it is more of a part of the timestamp verification, it should CAUSE the verification to fail, but not become the REASON to skip the check implicitly. Whether or not to perform the verification should be controlled by the trust policy instead of trust store.
3. In my PR, it states that the `level` field under `timestampVerification` only supports `strict` and `skip`. It performs like a boolean switch for timestamp verification: [Version 2.0](https://github.com/notaryproject/specifications/blob/2a6833fb66af44c3f974b2abaceedd6b543bcc49/specs/trust-store-trust-policy.md#version-20).

Given above, I'd like to list the use cases (as asked by @priteshbandi 
 in the community meeting) regarding timestamp verification (under `notary.x509` signing scheme) and how should the user config the trust policy accordingly:
```
# Use case 1: as a verifier, I'd like to verify timestamp from my trusted TSA, because my signing certificate has expired:
`Authentic timestamp` is `strict`. `level` under `timestamp verification` is also `strict`, timestamp verification is performed.

# Use case 2: I'd like to check signing cert expiration, but I don't trust any 3rd party TSA. As long as the signing cert has expired, fail the overall verification:
`Authentic timestamp` is `strict`. `level` under `timestamp verification` is `skip`, timestamp verification is always skipped. 
So, as you can see, user might accidentally include a `tsa` trust store (because they copy paste a trust policy from somewhere else and forgot to remove the trust store config), still NO timestamp verification is performed.

# Use case 3: I want to check signing cert expiration but only log it on failures:
`Authentic timestamp` is `permissive` or `audit`.  `level` under `timestamp verification` can be `strict` or `skip` depends on user's preferences. Only log on failures as required by `Authentic timestamp`.

# Use case 4: I just want to skip any signing cert expiration check (this is allowed in our already released v1.0.0 spec):
`Authentic timestamp` is `skip`. Any configuration of `timestamp verification` is ignored, and no timestamp verification is performed.
```

Reply:
> * `expiryRelaxed`: As per spec, we dont care about expiry of timestamping leaf cert, then why do we need this field?

We actually do care about it. Its default value is `false` meaning tsa certs need to be unexpired at the time of verification. This is following the `secure by default` strategy. So, it's more about security instead of usability.